### PR TITLE
Fix some docstrings and add validation testing for `deterministic.py`

### PR DESCRIPTION
**Pull request by dylanhmorris**

### CODE REVIEW

File: model/src/pyrenew/deterministic/deterministic.py [hunk 1]
```diff
@@ -63,10 +63,10 @@ def validate(value: ArrayLike) -> None:
         Raises
         ------
         Exception
-            If the input value object is not a ArrayLike.
+            If the input value object is not an ArrayLike object.
         """
         if not isinstance(value, ArrayLike):
-            raise Exception("value is not a ArrayLike")
+            raise ValueError("value is not an ArrayLike object")
```
Comment (damonbayer) [inline model/src/pyrenew/deterministic/deterministic.py:69]: Can we print the object and its type?

**Review by damonbayer — APPROVED** [review]
One optional improvement.

File: model/src/pyrenew/deterministic/deterministic.py (hunk 1 above)
Comment (dylanhmorris) [reply model/src/pyrenew/deterministic/deterministic.py]: addressed in 73711ce

**Review by damonbayer — APPROVED** [review]

File: model/src/pyrenew/deterministic/deterministic.py [hunk 2]
```diff
@@ -63,10 +63,14 @@ def validate(value: ArrayLike) -> None:
...
-            If the input value object is not a ArrayLike.
+            If the input value object is not an ArrayLike object.
         """
         if not isinstance(value, ArrayLike):
-            raise Exception("value is not a ArrayLike")
+            raise ValueError(
+                f"value {value} passed to a DeterministicVariable "
+                f"is of type {type(value).__name__}, expected "
+                "an ArrayLike object"
+            )
```
Comment (damonbayer) [inline model/src/pyrenew/deterministic/deterministic.py:73]: We have this type of check error throughout the project. Could be good to write a function for it.

**Pull request merged**